### PR TITLE
fix: Align popover inside dropdown

### LINKED ISSUE #788: Popover is always stretched to the right edge of the viewport
### Description
This happens on the documentation website as well as in my application that uses Fundamental-react.
image

### Versions
**fundamental-react:** current
**fundamental-styles:** none

---

_**NOTE:** Where applicable, please include uncropped screen captures._

_**DISCLAIMER:**
After triaging an issue, the fundamental-react team will see if it can be reproduced or confirmed. If more information is needed, the fundamental-react team will contact the author.  Any issues awaiting responses from the author for more than 7 days will be closed.  The author can re-open the issue at a later time if they can present the requested information._

### PR DESCRIPTION
**Pull request by parostatkiem**
#### This is a temporary solution.
The whole problem requires a major refactoring on popovers, as @jbadan said on Slack (100% agree).
This is just a "patch" to make Popover working with Dropdown or any other element.
⚠️ It **does not** introduce any backwards-incompatible changes.
### Description
- Add ability to align Popover left and right edge with its parent control element edges. 
- Add ability to set Popover right edge to be not-further-left or not-further-right than parent control right edge.

 It works well with Dropdown as well as any other type. ~Sadly, the solution requires creating a reference on the client side.~

#### Before:
image

#### After:
image

fixes #788

### CODE REVIEW

File: src/Popover/Popover.js [hunk 1]
```diff
@@ -90,22 +91,24 @@ Popover.propTypes = {
...
+    onClickOutside: () => { },
+    onEscapeKey: () => { }
 };
 
 Popover.propDescriptions = {
     body: 'Node(s) to render in the overlay.',
     control: 'Node to render as the reference element (that the `body` will be placed in relation to).',
     disableEdgeDetection: 'Set to **true** to render popover without edge detection so popover will not flip from top to bottom with scroll.',
     noArrow: 'Set to **true** to render a popover without an arrow.',
+    parentElement: 'A reference (useRef) to an element of which the Popover will be positioned and sized relatively. Leave it empty to let it be done automatically.',
```
Comment (markpalfreeman) [inline src/Popover/Popover.js]: Can we call this `parentRef` to be even more explicit?

And if we're going to mention `useRef()`, can we clarify the function itself shouldn't be passed in, but rather the return object from calling it?

File: src/utils/_Popper.js [hunk 2]
```diff
@@ -94,6 +95,11 @@ class Popper extends React.Component {
                     if (!show) {
                         return null;
                     }
+                    const currentParentElement = parentElement && parentElement.current;
```
Comment (markpalfreeman) [inline src/utils/_Popper.js]: Regarding my other comment, `parentRef` would be _slightly_ clearer here that we could access a property `.current`.

File: src/Popover/Popover.js (hunk 1 above)
Comment (markpalfreeman) [reply src/Popover/Popover.js]: Also, the description seems to imply ("let it be done") that positioning and sizing **relative to its parent** _can_ happen automatically if left empty. Is that the case?

**Comment by bcullman** [comment]
@parostatkiem - can you add before and after screen caps?

File: src/Popover/Popover.js (hunk 1 above)
Comment (parostatkiem) [reply src/Popover/Popover.js]: Sadly, it's not. If it was, this PR wouldn't need to exist at all ;)
I had no idea how to let the developer know he should put useRef's _result_ there... Your idea of `parentRef` seems reasonable.

I tried to put some comments into the code displayed in the documentation website but it's impossible :(

**Comment by parostatkiem** [comment]
@bcullman you can clearly see it live on the deployed preview.
https://deploy-preview-789--fundamental-react.netlify.com/dropdown

Open the _Width limited to parent's_ dropdown, then open any other one. You'll see the difference :)

**Comment by parostatkiem** [comment]
@jbadan ✅done

**Comment by bcullman** [comment]
@parostatkiem - I could imagine a better long term fix would be to address this here by adding a opt-in prop for `widthSizingType' here:

https://github.com/SAP/fundamental-react/blob/master/src/utils/_Popper.js

that accepted the following 

    ‘none’,
   ‘matchTarget’,
   ‘minTarget’,
   ‘maxTarget’

it would fix this component, and be flexible enough to handle all other circumstances, including other copmponets relying on `_popper.js`

**Comment by parostatkiem** [comment]
@bcullman - where it would take the target's width from? The problem is Popper has no "idea" of his parent. That's why I introduced `parentRef` which has an obvious disadvantage of being defined on the client side.

... or do you mean adding `widthSizingType` **on top of** my solution?

**Comment by greg-a-smith** [comment]
@parostatkiem The `_Popper.js` component has a prop called `referenceComponent`, which is the element to be sized in relation to.

**Comment by parostatkiem** [comment]
@greg-a-smith it was the first place I tried to find a solution at but I couldn't get the real DOM node of the `referenceComponent`, so I abandoned the idea and did it the other way.
However, today you inspired me to give it another chance and... here it is 🍾
Thank you for the inspiration!

@bcullman I implemented your solution in an unchanged form. I finally got rid of a ref on the client side :)

**Comment by jbadan** [comment]
@parostatkiem I tried to resolve some conflicts for you here, but I can't push to your branch to fix the failing test. Ping the team when the tests pass and we will merge :)

**Comment by parostatkiem** [comment]
@jbadan I added you to collaborators, you should receive an invitation email.
I'm quite busy this night (it's midnight right now) and will be able to fix the tests on Monday at the earliest.

**Comment by jbadan** [comment]
closing for #804

**Pull request closed without merging**